forte/models/da_rl/generator_trainer.py

```diff
@@ -0,0 +1,319 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A wrapper adding data augmentation to a model with arbitrary tasks.
+"""
+
+import random
+import texar.torch as tx
+from torch.nn import functional as F
+import torch
+
+from forte.models.da_rl.magic_model import MetaModule
+
+
+__all__ = [
+    "MetaAugmentationWrapper"
+]
+
+
+class MetaAugmentationWrapper:
+    # pylint: disable=line-too-long
+    r"""A wrapper adding data augmentation to a model with arbitrary tasks.
+
+    See: https://arxiv.org/pdf/1910.12795.pdf
+
+    There is an example code for this class here:
+    https://github.com/tanyuqian/learning-data-manipulation/blob/master/augmentation/generator.py
+
+    Let theta be the parameters of the downstream (classifier) model.
+    Let phi be the parameters of the augmentation model.
+    Equations to update phi:
+
+    theta'(phi) = theta - \nabla_{theta} L_{train}(theta, phi)
+
+    phi = phi - \nabla_{phi} L_{val}(theta'(phi))
+
+    Args:
+        augmentation_model:
+            A Bert-based language model for data augmentation.
+        augmentation_optimizer:
+            An optimizer.
```

Comment:
These comments do not provide additional information than the name itself, can you elaborate?